### PR TITLE
Add aptible_environment.backup_retention_policy and data source

### PR DESCRIPTION
**Pull request by joshraker**
- [x] Resource
- [x] Data Source
- [x] Examples & Docs

### CODE REVIEW

**Comment by joshraker** [comment]
Acceptance tests passing

```
❯ make testacc TESTARGS='-run BackupRetentionPolicy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -count 1 -parallel 12 -timeout 30m -run BackupRetentionPolicy
...
--- PASS: TestAccDataSourceBackupRetentionPolicy_validation (0.28s)
--- PASS: TestAccResourceBackupRetentionPolicy_validation (0.90s)
--- PASS: TestAccResourceBackupRetentionPolicy_basic (3.40s)
--- PASS: TestAccDataSourceBackupRetentionPolicy_basic (4.74s)
--- PASS: TestAccResourceBackupRetentionPolicy_update (5.20s)
PASS
ok      github.com/aptible/terraform-provider-aptible/aptible   5.765s
```

**Comment by joshraker** [comment]
Consider reviewing the last commit separately since it's changes to unrelated docs:

- https://github.com/aptible/terraform-provider-aptible/pull/114/files/2423e193444653342dff289a0d510c3e8574e3d2
- https://github.com/aptible/terraform-provider-aptible/pull/114/files/2423e193444653342dff289a0d510c3e8574e3d2..50247dd15bbc0e9bbde0e9080b589948fa19e6de

**Review by neurosnap — DISMISSED** [review]

**Comment by joshraker** [comment]
Acceptance tests passing after the move

```
❯ make testacc TESTARGS="-run 'TestAccResourceEnvironment|TestAccDataSourceBackupRetentionPolicy'"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -count 1 -parallel 12 -timeout 30m -run 'TestAccResourceEnvironment|TestAccDataSourceBackupRetentionPolicy'
...
--- PASS: TestAccDataSourceBackupRetentionPolicy_validation (1.01s)
--- PASS: TestAccResourceEnvironment_validation (1.42s)
--- PASS: TestAccDataSourceBackupRetentionPolicy_basic (3.87s)
--- PASS: TestAccResourceEnvironment_basic (4.20s)
--- PASS: TestAccResourceEnvironment_no_org (4.48s)
--- PASS: TestAccResourceEnvironment_backup_policy (4.48s)
--- PASS: TestAccResourceEnvironment_update (5.93s)
PASS
ok      github.com/aptible/terraform-provider-aptible/aptible   6.461s
```

**Comment by joshraker** [comment]
If multiple backup retention policies are specified, an error is raised before apply.

Screenshot 2024-07-02 at 16 19 53

File: docs/resources/environment.md [hunk 1]
```diff
@@ -33,3 +70,16 @@ Existing Environments can be imported using the Environment ID. For example:
...
+
+
+
+
+
+
+
+
+
+
```
Comment (neurosnap) [inline docs/resources/environment.md:84]: intentional?

**Review by neurosnap — APPROVED** [review]

File: docs/resources/environment.md (hunk 1 above)
Comment (joshraker) [reply docs/resources/environment.md]: Nope, thanks!

**Comment by joshraker** [comment]
```
❯ make testacc TESTARGS="-run 'TestAccResourceEnvironment|TestAccDataSourceBackupRetentionPolicy'"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -count 1 -parallel 12 -timeout 30m -run 'TestAccResourceEnvironment|TestAccDataSourceBackupRetentionPolicy'
...
--- PASS: TestAccDataSourceBackupRetentionPolicy_validation (0.46s)
--- PASS: TestAccResourceEnvironment_validation (0.92s)
--- PASS: TestAccResourceEnvironment_backup_policy (5.97s)
--- PASS: TestAccDataSourceBackupRetentionPolicy_basic (5.99s)
--- PASS: TestAccResourceEnvironment_basic (7.14s)
--- PASS: TestAccResourceEnvironment_no_org (7.26s)
--- PASS: TestAccResourceEnvironment_update (8.75s)
PASS
ok      github.com/aptible/terraform-provider-aptible/aptible   9.312s
```

**Review by neurosnap — APPROVED** [review]

**Pull request merged**